src/DataStructures/DataBox/Prefixes.hpp

```diff
@@ -14,32 +16,123 @@
+partial_derivatives(
+    const Variables<VariableTags>& u, const Index<Dim>& extents,
+    const Tensor<DataVector, tmpl::integral_list<std::int32_t, 2, 1>,
+                 typelist<SpatialIndex<Dim, UpLo::Up, Frame::Logical>,
+                          SpatialIndex<Dim, UpLo::Lo, DerivativeFrame>>>&
+        inverse_jacobian);
+/// \endcond
+
+namespace Tags {
+/*!
+ * \ingroup DataBoxTagsGroup
+ * \brief Prefix for a spatial derivative
+ *
+ * There are four variants of this tag that change how the derivatives are
+ * computed depending on which is chosen. The two simplest are the non-compute
+ * item versions for Tensor and Variables tags. These take three template
+ * parameters:
+ * 1. The tag to wrap
+ * 2. The volume dim as a type (e.g. `tmpl::size_t<Dim>`)
+ * 3. The frame of the derivative index in `Tensor`s
+ *
+ * The partial derivatives in a Frame that is not the logical frame as a compute
+ * item is requested by specifying the list of tags in the
+ * `Tags::Variables<tmpl::list<Tags...>>`, the list of tags to be
+ * differentiated, and the Tag for inverse Jacobian between the logical frame
+ * and the frame in which the derivative is taken. Note that the derivative Tags
+ * must be a contiguous subset from the head of the `Tags...` pack.
+ *
+ * The partial derivatives in the logical frame can also be computed. In that
+ * case the template parameters must be the list of Variables tags and
+ * derivative tags, with the last parameter being a `std::integral_constant` of
+ * the dimension of the mesh.
+ */
+template <typename Tag, typename VolumeDim, typename Frame, typename>
+struct deriv;
+
+template <typename Tag, typename VolumeDim, typename Frame>
+struct deriv<Tag, VolumeDim, Frame,
+             Requires<tt::is_a_v<Tensor, db::item_type<Tag>>>>
     : db::DataBoxPrefix {
   using type = TensorMetafunctions::prepend_spatial_index<
-      db::item_type<Tag>, VolumeDim::value, UpLo::Lo, Fr>;
+      db::item_type<Tag>, VolumeDim::value, UpLo::Lo, Frame>;
   using tag = Tag;
   static constexpr db::DataBoxString label = "d";
 };
 
-template <typename Tag, typename VolumeDim, typename Fr>
-struct d<Tag, VolumeDim, Fr,
-         Requires<tt::is_a_v<::Variables, db::item_type<Tag>>>>
+template <typename Tag, typename VolumeDim, typename Frame>
+struct deriv<Tag, VolumeDim, Frame,
+             Requires<tt::is_a_v<::Variables, db::item_type<Tag>>>>
     : db::DataBoxPrefix {
   using type = db::item_type<Tag>;
   using tag = Tag;
   static constexpr db::DataBoxString label = "d";
 };
 
+// Partial derivatives with derivative index in frame related to by
+// InverseJacobianTag
+template <typename... VariablesTags, typename... DerivativeTags,
+          typename InverseJacobianTag>
+struct deriv<tmpl::list<VariablesTags...>, tmpl::list<DerivativeTags...>,
+             InverseJacobianTag,
+             Requires<tt::is_a_v<Tensor, db::item_type<InverseJacobianTag>>>>
+    : db::ComputeItemTag {
```

Comment:
This looks like it will give us label conflicts as soon as we have multiple Variables or multiple frames.  I don't know what to do about it here, though, since the prefix tag stuff only uses one parameter.